### PR TITLE
resolve #66 in order to allow for determining the origin of a message

### LINKED ISSUE #66: Populate origin field of MessageHeaders
This comes from https://docs.celeryproject.org/en/latest/internals/reference/celery.utils.nodenames.html#celery-utils-nodenames

### PR DESCRIPTION
**Pull request by RestitutorOrbis**

### CODE REVIEW

File: src/protocol/mod.rs [hunk 1]
```diff
@@ -24,6 +26,18 @@ pub struct MessageBuilder {
...
+        let pid = process::id().to_string();
+        let prefix = "gen";
+        let name = prefix.to_owned() + &pid;
+        let system_hostname_result = hostname::get();
+        let system_hostname = match system_hostname_result {
+            Ok(system_hostname_ok) => system_hostname_ok.into_string(),
+            Err(err) => panic!(err),
+        };
+        let origin = match system_hostname {
+            Ok(system_hostname_string) => name + "@" + &system_hostname_string,
```
Comment (epwalsh) [inline src/protocol/mod.rs]: You may want to try `=> format!("{}@{}", name, system_hostname_string)`

**Comment by epwalsh** [comment]
@RestitutorOrbis just thought I'd ping you to see if you're planning on continuing with this. No worries if not. Hope you're staying healthy.

**Comment by RestitutorOrbis** [comment]
@epwalsh Thanks for pinging me! Sorry, I've been busy with school. I'll hopefully get to it this weekend. Hope you're staying healthy too!

File: src/protocol/mod.rs [hunk 2]
```diff
@@ -31,6 +33,18 @@ where
...
+        let name = prefix.to_owned() + &pid;
+        let system_hostname_result = hostname::get();
+        let system_hostname = match system_hostname_result {
+            Ok(system_hostname_ok) => system_hostname_ok.into_string(),
+            Err(err) => panic!(err),
+        };
+        let origin = match system_hostname {
+            Ok(system_hostname_string) => format!("{}@{}", name, system_hostname_string),
+            Err(err) => panic!(err),
+        };
```
[the remark this answers is not in the document]
Comment (RestitutorOrbis) [reply src/protocol/mod.rs]: I adopted that suggestion and I agree with you wrt the use of lazy initialization. It's all been implemented in my latest revision.

**Pull request merged**